spec/rubocop/cop/style/collection_compact_spec.rb

```diff
@@ -199,11 +177,17 @@ def foo(params)
     RUBY
   end
 
+  it 'does not register an offense when using `delete_if`' do
```

Comment:
It may be helpful to add a comment like the following:

```suggestion
  # NOTE: Unlike `reject` or `compact` which return a new collection object,
  # `delete_if` always returns its own object `self` making them incompatible.
  it 'does not register an offense when using `delete_if`' do
```

Reply:
Thank you for the review! I've made the changes.